distributed/scheduler.py

```diff
@@ -2119,11 +2119,13 @@ def decide_worker_rootish_queuing_disabled(
         tg.last_worker = (
             ws if tg.states["released"] + tg.states["waiting"] > 1 else None
         )
-        tg.last_worker_tasks_left -= 1
+        if tg.last_worker_tasks_left > 0:
+            tg.last_worker_tasks_left -= 1
```

Comment:
To my understanding, this covers the use case where
`(len(tg) / self.total_nthreads) * ws.nthreads` < 1.
Since a task is rootish iff `len(tg) > total_nthreads * 2`, this should be mathematically impossible.

Reply:
> Since a task is rootish iff `len(tg) > total_nthreads * 2`

The main change in this PR is that this is no longer the case. Since we now call any task with 0 dependents root-ish—regardless of the size of its TaskGroup—it's possible for `len(tg)` to be as small as 1.

Reply:
Ah, I see. I refactored the code to add clarity.